packages/bundler-plugin-core/src/debug-id-upload.ts

```diff
@@ -114,61 +113,62 @@ function addDebugIdToBundleSource(bundleSource: string, debugId: string): string
-  if (sourceMappingUrlMatch) {
-    const sourceMappingUrl = path.normalize(sourceMappingUrlMatch[1] as string);
+  const sourceMappingUrl = sourceMappingUrlMatch ? sourceMappingUrlMatch[1] as string : undefined;
 
-    let isUrl;
-    let isSupportedUrl;
+  const searchLocations: string[] = [];
+
+  if (resolveSourceMapHook) {
+    const customPath = await resolveSourceMapHook(bundlePath, sourceMappingUrl);
+    if (customPath) {
+      searchLocations.push(customPath);
+    }
+  }
+
+  // 1. try to find source map at `sourceMappingURL` location
+  if (sourceMappingUrl) {
+    let parsedUrl: URL | undefined;
     try {
-      const url = new URL(sourceMappingUrl);
-      isUrl = true;
-      isSupportedUrl = url.protocol === "file:";
+      parsedUrl = new URL(sourceMappingUrl);
     } catch {
-      isUrl = false;
-      isSupportedUrl = false;
+      // noop
     }
 
-    let absoluteSourceMapPath;
-    if (isSupportedUrl) {
-      absoluteSourceMapPath = sourceMappingUrl;
-    } else if (isUrl) {
-      // noop
+    if (parsedUrl && parsedUrl.protocol === "file:") {
+      searchLocations.push(url.fileURLToPath(sourceMappingUrl));
+    } else if (parsedUrl) {
+      // noop, non-file urls don't translate to a local sourcemap file
     } else if (path.isAbsolute(sourceMappingUrl)) {
-      absoluteSourceMapPath = sourceMappingUrl;
+      searchLocations.push(path.normalize(sourceMappingUrl))
     } else {
-      absoluteSourceMapPath = path.join(path.dirname(bundlePath), sourceMappingUrl);
-    }
-
-    if (absoluteSourceMapPath) {
-      try {
-        // Check if the file actually exists
-        await util.promisify(fs.access)(absoluteSourceMapPath);
-        return absoluteSourceMapPath;
-      } catch (e) {
-        // noop
-      }
+      searchLocations.push(path.normalize(path.join(path.dirname(bundlePath), sourceMappingUrl)));
     }
   }
 
   // 2. try to find source map at path adjacent to chunk source, but with `.map` appended
-  try {
-    const adjacentSourceMapFilePath = bundlePath + ".map";
-    await util.promisify(fs.access)(adjacentSourceMapFilePath);
-    return adjacentSourceMapFilePath;
-  } catch (e) {
-    // noop
+  searchLocations.push(bundlePath + ".map")
+
+  for (const searchLocation of searchLocations) {
+    try {
+      await util.promisify(fs.access)(searchLocation);
+      logger.debug(`Source map found for bundle \`${bundlePath}\`: \`${searchLocation}\``)
+      return searchLocation;
+    } catch (e) {
+      // noop
+    }
   }
 
   // This is just a debug message because it can be quite spammy for some frameworks
   logger.debug(
-    `Could not determine source map path for bundle: ${bundlePath} - Did you turn on source map generation in your bundler?`
+    `Could not determine source map path for bundle: \`${bundlePath}\`` +
+    ` - Did you turn on source map generation in your bundler?` +
+    ` (Attempted paths: ${searchLocations.map(e => `\`${e}\``).join(", ")})`
```

Comment:
low key love this change.

Reply:
While I was addressing the other comments I made it so this logging statement also includes the `sourceMappingURL` value, so it should be clearer why it might be searching in specific locations